libr/core/cmd_print.c

```diff
@@ -4840,6 +4840,38 @@ static int cmd_print(void *data, const char *input) {
 		case 'J': // pdJ
 			formatted_json = true;
 			break;
+		case 'u': // pdu
+			processed_cmd = true;
+			if (input[2] == '?') {
+				//todo - print help for pdu/pdub/pdur
+			} else if (input[2] == 'b' ){ // pdub
+				//todo - define pdub
+			} else if (input[2] == 'r') { //pdur
+				//todo - define pdur
+			} else { // pdu
+				ut64 start_addr = core->offset;
+				ut8 *buf = NULL;
+				if (strlen(input) < 3){
+					eprintf ("Invalid address ranges\n");
+					break;
+				}
+				ut64 end_addr = r_num_math (core->num, (char *)(input + 3));
+				l = end_addr - start_addr;
+				const int bs = core->blocksize;
+				int bs1 = l * 16;
+				int bsmax = R_MAX (bs, bs1);
+				buf = malloc (bsmax + 1);
+				if (buf) {
+					memcpy (buf, block, bs);
+					if (bs1 > bs) {
+						r_io_read_at (core->io, start_addr + bs / addrbytes, buf + (bs - bs % addrbytes),
+								bs1 - (bs - bs % addrbytes));
+					}
+					core->num->value = r_core_print_disasm (core->print, core, start_addr, buf, l, l, 0, 1, 0, NULL);
```

Comment:
the disassembling of pdu should not be linear, it implies esil emulation and must follow branches, otherwise its the same as doing pD Y-X@X